packages/core/src/scuttle.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * @typedef {object} ScuttleOpts
+ * @property {boolean} enabled - Whether scuttling is enabled or not.
+ * @property {Array<string|RegExp>} exceptions - List of properties to exclude from scuttling.
+ * @property {string} scuttlerName - Name of the scuttler function to use.
+ * @property {(globalRef: object, scuttle: Function) => void} scuttlerFunc - Scuttler function to use.
+ */
+
+/**
+ * @typedef {object} GlobalRef
+ * @property {object} [globalThis] - Reference to the global object.
```

Comment:
If `globalThis` is intended to be read-only for consumers of this code, then the type `object` is OK enough. If we intend to write to this property, then it should be `Record<string, any>`

Reply:
It's meant to have some of its properties deleted, so I should probably switch to a Record.